From 41bdfe6db9655aae1131d31e4aa4b6337a1e268d Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Wed, 18 Sep 2019 15:20:00 +0200
Subject: [PATCH 04/24] Emscripten doesn't implement `pthread_getname_np()`

Upstream-Status: Inappropriate [Wasm-specific]
GLib will not integrate any Wasm-specific patches until the toolchain
supports calling functions through differently typed function pointers,
see: https://gitlab.gnome.org/GNOME/glib/-/issues/3596

(cherry picked from commit 8ced885eb25b0b931f3ac9690bd1804910619e80)
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 4b80c6df0..3c7ae2df8 100644
--- a/meson.build
+++ b/meson.build
@@ -2160,7 +2160,7 @@ else
   if cc.has_header_symbol('pthread.h', 'pthread_cond_timedwait_relative_np')
     glib_conf.set('HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP', 1)
   endif
-  if cc.has_header_symbol('pthread.h', 'pthread_getname_np', prefix : pthread_prefix)
+  if host_system != 'emscripten' and cc.has_header_symbol('pthread.h', 'pthread_getname_np', prefix : pthread_prefix)
     glib_conf.set('HAVE_PTHREAD_GETNAME_NP', 1)
   endif
   if cc.has_header_symbol('pthread.h', 'pthread_getaffinity_np', prefix : pthread_prefix)
